Map/Source/SquareMap.cpp

```diff
@@ -6,36 +6,127 @@
 #include <memory>
 #include <stdexcept>
 #include <vector>
+#include <sstream>
 
 const FieldList& SquareMap::getFields()
 {
     return _fieldList;
 }
 
-char SquareMap::printField(const Field& field)
+char SquareMap::printField(const std::unique_ptr<Field>& field)
 {
-  if(field.isVisible())
-  {
-    FieldType type = field.getType();
-    switch (type) {
-      case FieldType::Empty:
-        return '.';
-      case FieldType::Wall:
-        return '#';
-      case FieldType::Fight:
-        return 'F';
-      case FieldType::Treasure:
-        return 'T';
-      case FieldType::Door:
-        return 'D';
-      default:
-        return '?';
+    if(field->isVisible())
+    {
+        FieldType type = field->getType();
+        switch (type) {
+            case FieldType::Empty:
+                return '.';
+            case FieldType::Wall:
+                return '#';
+            case FieldType::Fight:
+                return 'F';
+            case FieldType::Treasure:
+                return 'T';
+            case FieldType::Door:
+                return 'D';
+            default:
+                return '?';
+        }
+    }
+    else
+    {
+        return ' ';
+    }
+}
+
+std::string SquareMap::getMapToPrint(const Position& playerPosition)
+{
+    std::stringstream output;
+    
+    const unsigned int mapSize = _fieldList.size();
+    
+    printColumnNumbers(output, mapSize);
```

Comment:
Parametry in/out nie są dobrym rozwiązaniem. 
1. Funkcja nie wykorzystuje danych zawartych w parametrze output, wiec nie jest on tutaj potrzebny
2. Nie jestem pewny co dzieje sie ze zmnienna output w wnetrzu tej funkcji. Muszę sprawdzić implementacje, zeby dowiedziec sie o co chodzi.

Teraz co do nazewnictwa, printColumnNumbers a nic nie printujemy :P 
W tym wypadku powinno się to raczej nazywać setColumnNumberToPrint.

Ale z racji tego, że nie chcemy korzystać z tego parametru in/out jakim jest output, zrobmy cos takiego:
```
std::ostream SquareMap::getColumnNumbersToPrint(const unsigned int mapSize)
{
   std::ostream str;
    str << "   ";
    for(unsigned int ite = 0; ite < mapSize; ++ite)
    {
        str << ' ' << (ite + 1) % 10;
    }
    str << '\n';

  return str;
}
```

Reply:
i poźniej w wywołaniach tutaj
`output << getColumnNumbersToPrint(mapSize)`

Reply:
Te uwagi dotycza sie wszystkich funkcji z rodziny print....